guides/common/modules/proc_enabling-pull-based-transport.adoc

```diff
@@ -0,0 +1,40 @@
+[id="enabling-pull-based-tranport_{context}"]
+= Enabling Pull-based Transport
+
+.Procedure
+If your {Project} or {SmartProxy} is preinstalled, enable the pull provider using:
+ifdef::satellite[]
+`satellite-installer --foreman-proxy-plugin-remote-execution-script-mode=pull-mqtt`
+endif::[]
+`foreman-installer --foreman-proxy-plugin-remote-execution-script-mode=pull-mqtt`.
+
+You may additionally need to run `--foreman-proxy-templates=true --foreman-proxy-registration=true` to enable the {SmartProxy} for host registration.
+
+When the installer command succeeds, the MQTT broker runs at Port 1883.
+You must allow that port in your firewall by running the following commands:
+
+. `firewall-cmd --add-port="1883/tcp"`
+. `firewall-cmd --runtime-to-permanent`
+
+Choose one of three options to enable hosts to work with MQTT for pull provider:
+
+* Migrate existing clients to pull provider
+
+ifdef::satellite[]
+If your client is already registered to {Project}, you can convert the client to the pull-based transport if the client has access to the {Project} Client repository.
+
+Do this by following these steps:
+.Install the conversion tool by running `yum install katello-pull-transport-migrate`, which installs `foreman_ygg_worker` and `yggdrasil` as dependencies and performs the conversion.
+.To verify the installation was successful, run `ygdrassil status` and execute a remote execution job against the host using the script provider job.
+
+To register your host with the pull client configuration option enabled, navigate to the advanced fields of the *Global Registration Form* and set `REX pull mode` to `Yes`. 
+If you make no changes and leave the field on default, the value remains `host_registration_remote_execution_pull parameter`.
+
+Additionally, set the `enable-remote-execution-pull` parameter to `true` so the script from `katello-pull-transport-migrate` is put on the disk and executed during provisioning.
```

Comment:
This section belongs under *Provision clients to connect to pull provider*. It sounds like it was something to do in addition to the previous steps but it's actually an alternative. Mentioning `katello-pull-transport-migrate` is confusing a bit: there is a provisioning template called `remote_execution_pull_setup` that does the job -- it contains the same code as k-p-t-m but it is not the same thing.